### DISCUSSION TITLE
How to specify languages of delivery of the session?

### DISCUSSION THREAD

**Discussion by joffinjoy** [beckn/protocol-specifications#339] [Q&A]
According to our earlier understanding, we would be specifying **languages of delivery** of a session in its fulfilment. Meaning; a session could have multiple slots in different languages at different times. Something like this:

```
{
	"catalog": {
		"descriptor": {},
		"providers": [{
			"fulfillments": [{
				"languages": [{
					"code": "HI",
					"name": "Hindi"
				}, {
					"code": "EN",
					"name": "English"
				}]
			}],
			"categories": [{}],
			"items": []
		}]
	}
}
```
The above schema specifies that a session will be delivered in both Hindi and English during a single fulfilment. Like that, this **session** can have other **fulfilments** with other set of languages. 

But current schema doesn't have this **languages** field and hence doesn't satisfy above requirement. So how do we specify **languages of delivery** of a session? Can we add **languages** schema to root of **fulfilment**?

Note: Just to clarify, we did add a **languages** field to **Person** schema (Here: https://github.com/beckn/protocol-specifications/pull/327), but that simply indicates that the person knows those languages and isn't directly related to the sessions being taken by that person. 

cc: @ravi-prakash-v, @aks30, @vijiurs, @kiranharidas187

**Comment by sankarshanmukhopadhyay** [comment]
An observation (unrelated to the topic raised) - keeping the language codes similar/consistent with the Unicode CLDR might be useful in the long run when considering i18n readiness.

**Reply by joffinjoy** [reply]
I agree. The current codes and names are just placeholders until we adopt a network-wide standard for languages. Thanks for the response.
